curp/src/server/raw_curp/mod.rs

```diff
@@ -147,6 +148,8 @@ struct Context<C: Command> {
     sync_events: HashMap<ServerId, Arc<Event>>,
     /// Become leader event
     leader_event: Arc<Event>,
+    /// Leader change callback
+    role_change: Box<dyn RoleChange>,
```

Comment:
Is static generic possible?

Reply:
The `impl Trait` is only allowed in function and inherent method return types, not in path.

Reply:
Done